src/pipx/package_specifier.py

```diff
@@ -161,7 +161,24 @@ def parse_specifier_for_install(package_spec: str, pip_args: List[str]) -> Tuple
         )
         pip_args.remove("--editable")
 
-    return (package_or_url, pip_args)
+    pip_args = [option for pip_arg in pip_args for option in pip_arg.split(" ")]
+
+    for option in pip_args:
+        if "-c" in option or "--constraint" in option:
```

Comment:
Small refactoring suggestion (don't apply, lacks proper indentation):

```suggestion
    if any(arg.startswith("--constraint") for arg in pip_args):
```

Reply:
what about `"-c"`?

Reply:
True, updated the suggestion.

Reply:
done, but changed the inner block a little bit (as wrote above about the `-c` flag, please take a look :)

and again, thank you!